### PR TITLE
refactor: structure and components

### PR DESCRIPTION
**Pull request by songwongtp**
## Describe your changes
- Refactor various components to reduce redundancy
- Reorganize components from a specific page to the general component directory if used multiple locations

### CODE REVIEW

**Review by traviolus — APPROVED** [review]

File: src/lib/components/table/transactions/TransactionsTableRow.tsx [hunk 1]
```diff
@@ -1,49 +1,55 @@
...
-export const TxsTableRow = ({
+export const TransactionsTableRow = ({
   transaction,
   templateColumns,
-}: TxsTableRowProps) => {
+  showSender,
+}: TransactionsTableRowProps) => {
   const { isOpen, onToggle } = useDisclosure();
-  const isAccordion = transaction.messages.length > 1;
+  const [isAccordion, setIsAccordion] = useState(false);
```
Comment (evilpeach) [inline src/lib/components/table/transactions/TransactionsTableRow.tsx:32]:
```suggestion
  const isAccordion = transaction.messages.length > 1;
```

File: src/lib/model/contract.ts [hunk 2]
```diff
@@ -207,3 +205,20 @@ export const useContractDetailsTableCounts = (
     refetchRelatedProposals,
   };
 };
+
+export const useCodeContracts = (
```
Comment (evilpeach) [inline src/lib/model/contract.ts:209]: useContractsByCodeId

File: src/lib/pages/account-details/components/tables/AdminContractsTable.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    offset,
+  } = usePaginator({
+    total: totalData,
+    initialState: {
+      pageSize: 10,
+      currentPage: 1,
+      isDisabled: false,
+    },
+  });
+  const { contracts, isLoading } = useContractsAdmin(
```
Comment (evilpeach) [inline src/lib/pages/account-details/components/tables/AdminContractsTable.tsx:41]: useAccountAdminContracts

File: src/lib/pages/account-details/components/tables/InstantiatedContractsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    offset,
+  } = usePaginator({
+    total: totalData,
+    initialState: {
+      pageSize: 10,
+      currentPage: 1,
+      isDisabled: false,
+    },
+  });
+  const { contracts, isLoading } = useContractInstances(
```
Comment (evilpeach) [inline src/lib/pages/account-details/components/tables/InstantiatedContractsTable.tsx:41]: useAccountContracts

File: src/lib/pages/code-details/components/table/CodeContractsTable.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    return (
+      <>
+        <TableTitle title="Contract Instances" count={totalData ?? 0} />
+        <ContractsTable
+          contracts={codeContracts}
+          isLoading={isLoading}
+          emptyState={<NoContracts />}
+        />
+        {totalData && totalData > 10 && (
```
Comment (evilpeach) [inline src/lib/pages/code-details/components/table/CodeContractsTable.tsx:68]: !!totalData

File: src/lib/pages/code-details/components/table/CodeContractsTable.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    };
+
+    const onPageSizeChange = (e: ChangeEvent<HTMLSelectElement>) => {
+      const size = Number(e.target.value);
+      refetch();
+      setPageSize(size);
+    };
+
+    // FIXME - might be a better way to scroll to table header
+    const tableHeaderId = "contractTableHeader";
```
Comment (evilpeach) [inline src/lib/pages/code-details/components/table/CodeContractsTable.tsx:58]: use this on table header

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**